rflx/pyrflx/typevalue.py

```diff
@@ -492,42 +504,69 @@ class MessageValue(TypeValue):
 
     _type: Message
 
-    def __init__(self, model: Message, refinements: Sequence[Refinement] = None) -> None:
+    def __init__(
+        self,
+        model: Message,
+        refinements: Sequence[Refinement] = None,
+        skip_verification: bool = False,
+        fields: Mapping[str, "MessageValue.Field"] = None,
+        checksums: Mapping[str, "MessageValue.Checksum"] = None,
+        type_literals: Mapping[Name, Expr] = None,
+    ) -> None:
+        # pylint: disable=too-many-arguments
         super().__init__(model)
+        self._skip_verification = skip_verification
         self._refinements = refinements or []
-        self._fields: Dict[str, MessageValue.Field] = {
+
+        self._fields: Mapping[str, MessageValue.Field] = fields if fields else {
             f.name: self.Field(
                 TypeValue.construct(
-                    self._type.types[f], imported=self._type.types[f].package != model.package
+                    self._type.types[f] if f in self._type.types else Opaque(),
+                    imported=f in self._type.types and self._type.types[f].package != model.package,
                 ),
                 f.name,
             )
-            for f in self._type.fields
+            for f in (INITIAL,) + self._type.fields
```

Comment:
Why is a field for `INITIAL` added now?

Reply:
To ensure that `self._fields` cannot be changed after its creation (which would invalidate some assumptions) I changed its type to `Mapping`. Since a `Mapping` is immutable I have to add `INITIAL` in the dict comprehension.